internal/handlers/common/filter.go

```diff
@@ -452,3 +459,47 @@ func filterFieldExprBitsAnySet(fieldValue, maskValue any) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldMod handles {field: {$mod: [divisor, remainder]}} filter.
+func filterFieldMod(fieldValue, exprValue any) (bool, error) {
+	arr := exprValue.(*types.Array)
+	if arr.Len() < 2 {
+		return false, NewErrorMsg(ErrBadValue, "malformed mod, not enough elements")
+	}
+	if arr.Len() > 2 {
+		return false, NewErrorMsg(ErrBadValue, "malformed mod, too many elements")
+	}
+
+	div, err := toInt64(must.NotFail(arr.Get(0)))
+	if err != nil {
+		return false, NewErrorMsg(ErrBadValue, "malformed mod, divisor not a number")
+	}
+	rem, err := toInt64(must.NotFail(arr.Get(1)))
+	if err != nil {
+		return false, NewErrorMsg(ErrBadValue, "malformed mod, remainder not a number")
+	}
+	field, err := toInt64(fieldValue)
+	if err != nil {
+		return false, nil
+	}
+
+	if field%div != rem {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// toInt64 ...
+func toInt64(val any) (int64, error) {
```

Comment:
I'm not sure, but you might want to use existing `GetWholeNumberParam` function

Reply:
There is a condition in the `GetWholeNumberParam` function
```
if value != math.trunc(value) || math.IsNaN(value) || math.IsInf(value, 0) {
   return 0, errNotWholeNumber
}
```
float, for example 234.5, the function does not consider this value as a number

The `$mod` operator behaves a little differently - the float value is rounded up to 234.5 -> 234.

Reply:
We could call it `getModParam` then and move to `internal/handlers/common/params.go`